src/pipeline/src/etl/transform/mod.rs

```diff
@@ -38,6 +39,34 @@ pub trait Transformer: std::fmt::Display + Sized + Send + Sync + 'static {
     fn transform(&self, val: crate::etl::value::Value) -> Result<Self::Output, String>;
 }
 
+#[derive(Debug, Clone, Default)]
+pub enum OnFailure {
+    #[default]
+    Ignore,
+    Default,
+}
+
+impl std::str::FromStr for OnFailure {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
```

Comment:
to_lowercase?

Reply:
The config is yml-style. It will complain at compile(insertion) time if user gives a wrong key name. I think we are good for now.